pages/api/auth/[...nextauth].ts

```diff
@@ -1,70 +1,135 @@
-
-const {serverRuntimeConfig} = getConfig();
-
-export default NextAuth({
-  providers: [
-    GithubProvider({
-      clientId: serverRuntimeConfig?.github?.clientId,
-      clientSecret: serverRuntimeConfig?.github?.secret,
-      authorization:
-        "https://github.com/login/oauth/authorize?scope=read:user+user:email+repo",
-      profile(profile: { id; name; login; email; avatar_url }) {
-        return {
-          id: profile.id,
-          name: profile.name,
-          login: profile.login,
-          email: profile.email,
-          image: profile.avatar_url,
-        };
-      },
-    }),
-  ],
-  pages: {
-    signIn: "/auth/signin"
-  },
-  session:{
-    strategy: "jwt",
-    maxAge: 30 * 24 * 60 * 60 // 30 days
-  },
-  callbacks: {
-    async signIn({ profile }) {
-      try {
-        if (!profile?.login) return "/?authError=Profile not found";
-
-        return true;
-      } catch(e) {
-        console.log("SignIn Callback", e);
-        return false;
-      }
-    },
-    async jwt({ token, account, profile }) {
-      const user = await models.user.findOne({
-        where: { 
-          githubLogin: profile?.login || token?.login
-        },
-        raw: true
-      })
-        .catch(error => {
-          console.log("JWT Callback", error)
-        });
-
-      return { ...token, ...profile, ...account, wallet: user?.address };
-    },
-    async session({ session, token }) {
-      // console.log(`Session`, session, user, token);
-      return {
-        expires: session.expires,
-        user: {
-          ...session.user,
-          login: token.login,
-          accessToken: token?.access_token,
-        },
-      };
-    },
-  },
-});
+import { NextApiRequest, NextApiResponse } from "next";
+import NextAuth, { Account, Profile } from "next-auth";
+import { getToken } from "next-auth/jwt";
+import getConfig from "next/config";
+
+import models from "db/models";
+
+import { DAY_IN_SECONDS } from "helpers/constants";
+import { caseInsensitiveEqual } from "helpers/db/conditionals";
+import { toLower } from "helpers/string";
+
+import { Logger } from "services/logging";
+
+import { EthereumProvider, GHProvider } from "server/auth/providers";
+
+const {
+  publicRuntimeConfig,
+  serverRuntimeConfig: {
+    auth: {
+      secret
+    }
+  }
+} = getConfig();
+
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+  const currentToken = await getToken({ req, secret: secret });
+
+  return NextAuth(req, res, {
+    providers: [
+      EthereumProvider,
+      GHProvider
+    ],
+    pages: {
+      signIn: "/auth/signin"
+    },
+    session:{
+      strategy: "jwt",
+      maxAge: 30 * DAY_IN_SECONDS
+    },
+    callbacks: {
+      async signIn({ profile, account}) {
+        try {
+          const provider = account?.provider;
+
+          if (provider === "github") {
+            if (!profile?.login) return "/?authError=Profile not found";
+
+            return true;

```

Comment:
I'm not sure we should return true here; either we leave the same implementation we had until we play the "remove connect github account" or we return false -- or completly remove the "GithubProvider" altogether because we stop supporting "logging in with github"